packages/office-ui-fabric-react/src/components/Coachmark/Coachmark.base.tsx

```diff
@@ -376,19 +394,30 @@ export class CoachmarkBase extends BaseComponent<ICoachmarkProps, ICoachmarkStat
     const { preventDismissOnLostFocus } = this.props;
     const currentDoc: Document = getDocument()!;
 
-    this._events.off();
-
     if (currentDoc) {
-      this._events.on(currentDoc, 'keydown', this._onKeyDown, true);
-
+      this._async.setTimeout(() => {
+        this._disposables.push(on(currentDoc.documentElement, 'keydown', this._onKeyDown, true));
+        this._hasListeners = true;
+      }, 0);
       if (!preventDismissOnLostFocus) {
-        this._events.on(currentDoc, 'click', this._dismissOnLostFocus, true);
-        this._events.on(currentDoc, 'focus', this._dismissOnLostFocus, true);
+        this._async.setTimeout(() => {
+          this._disposables.push(
+            on(currentDoc.documentElement, 'click', this._dismissOnLostFocus, true),
+            on(currentDoc.documentElement, 'focus', this._dismissOnLostFocus, true)
+          );
+          this._hasListeners = true;
+        }, 0);
       }
     }
   }
 
-  private _dismissOnLostFocus(ev: Event) {
+  private _removeListeners() {
+    this._disposables.forEach((dispose: () => void) => dispose());
```

Comment:
Should removal occur when `addListeners` is called as it was prior to these changes?

Reply:
this._hasListeners check ensures that removal only occurs if there are existing listeners already - I think removing inside addition is a bit confusing...